### PR TITLE
Fix docs for new plotting classes

### LINKED ISSUE #1307: Doc builds broken
@fidelram @gokceneraslan I'm opening this issue to talk about the broken doc builds. Previously discussed in: https://github.com/theislab/scanpy/pull/1204#issuecomment-654765480 and https://github.com/theislab/scanpy/pull/1210#issuecomment-651510328

I'm able to see the errors, but that might be due to privileges for readthedocs. Are you able to look at any PR error logs?

Anyways, the docs fail with this traceback:

<details>
<summary> Traceback </summary>

```

/home/docs/checkouts/readthedocs.org/user_builds/icb-scanpy/envs/1204/lib/python3.6/site-packages/setuptools_scm/git.py:68: UserWarning: "/home/docs/checkouts/readthedocs.org/user_builds/icb-scanpy/checkouts/1204" is shallow and may cause errors
  warnings.warn('"{}" is shallow and may cause errors'.format(wd.path))
/home/docs/checkouts/readthedocs.org/user_builds/icb-scanpy/checkouts/1204/docs/conf.py:112: RemovedInSphinx40Warning: The app.add_stylesheet() is deprecated. Please use app.add_css_file() instead.
  app.add_stylesheet('css/custom.css')

Traceback (most recent call last):
  File "/home/docs/checkouts/readthedocs.org/user_builds/icb-scanpy/envs/1204/lib/python3.6/site-packages/sphinx/cmd/build.py", line 280, in build_main
    app.build(args.force_all, filenames)
  File "/home/docs/checkouts/readthedocs.org/user_builds/icb-scanpy/envs/1204/lib/python3.6/site-packages/sphinx/application.py", line 348, in build
    self.builder.build_update()
  File "/home/docs/checkouts/readthedocs.org/user_builds/icb-scanpy/envs/1204/lib/python3.6/site-packages/sphinx/builders/__init__.py", line 299, in build_update
    len(to_build))
  File "/home/docs/checkouts/readthedocs.org/user_builds/icb-scanpy/envs/1204/lib/python3.6/site-packages/sphinx/builders/__init__.py", line 311, in build
    updated_docnames = set(self.read())
  File "/home/docs/.pyenv/versions/3.6.8/lib/python3.6/contextlib.py", line 88, in __exit__
    next(self.gen)
  File "/home/docs/checkouts/readthedocs.org/user_builds/icb-scanpy/envs/1204/lib/python3.6/site-packages/sphinx/util/logging.py", line 213, in pending_warnings
    memhandler.flushTo(logger)
  File "/home/docs/checkouts/readthedocs.org/user_builds/icb-scanpy/envs/1204/lib/python3.6/site-packages/sphinx/util/logging.py", line 178, in flushTo
    logger.handle(record)
  File "/home/docs/.pyenv/versions/3.6.8/lib/python3.6/logging/__init__.py", line 1454, in handle
    self.callHandlers(record)
  File "/home/docs/.pyenv/versions/3.6.8/lib/python3.6/logging/__init__.py", line 1516, in callHandlers
    hdlr.handle(record)
  File "/home/docs/.pyenv/versions/3.6.8/lib/python3.6/logging/__init__.py", line 861, in handle
    rv = self.filter(record)
  File "/home/docs/.pyenv/versions/3.6.8/lib/python3.6/logging/__init__.py", line 720, in filter
    result = f.filter(record)
  File "/home/docs/checkouts/readthedocs.org/user_builds/icb-scanpy/envs/1204/lib/python3.6/site-packages/sphinx/util/logging.py", line 415, in filter
    raise SphinxWarning(location + ":" + str(message))
sphinx.errors.SphinxWarning: /home/docs/checkouts/readthedocs.org/user_builds/icb-scanpy/checkouts/1204/scanpy/plotting/_dotplot.py:docstring of scanpy.pl.dotplot:122:Inline strong start-string without end-string.

Warning, treated as error:
/home/docs/checkouts/readthedocs.org/user_builds/icb-scanpy/checkouts/1204/scanpy/plotting/_dotplot.py:docstring of scanpy.pl.dotplot:122:Inline strong start-string without end-string.
```

</details>

I get the same error locally from just running `make html`, and was able to get rid of it by adding a page for the `DotPlot` class (though more errors came up after that).

### PR DESCRIPTION
**Pull request by fidelram**
Fixes #1307

### CODE REVIEW

**Comment by ivirshup** [comment]
I think the main issue with the docs building right now is that there are references to the `DotPlot` class, but there isn't any page built for those to link to.

I'm not sure the `**kwargs` argument is causing an issue, since the docs for `sc.queries.enrich` have this, and seem to work fine: https://scanpy.readthedocs.io/en/stable/api/scanpy.queries.enrich.html

I have had issues with doc build errors failing to show up if you just try and build the docs again. It might be worth running `make clean` before each `make build` to see what happens.

File: docs/conf.py [hunk 1]
```diff
@@ -137,6 +137,9 @@ def setup(app):
     "sklearn.neighbors.dist_metrics.DistanceMetric": "sklearn.neighbors.DistanceMetric",
     # If the docs are built with an old version of numpy, this will make it work:
     "numpy.random.RandomState": "numpy.random.mtrand.RandomState",
+    "scanpy.plotting._matrixplot.MatrixPlot": "scanpy.plotting.MatrixPlot",
+    "scanpy.plotting._dotplot.DotPlot": "scanpy.plotting.DotPlot",
+    "scanpy.plotting._stacked_violin.StackedViolin": "scanpy.plotting.StackedViolin",
```
Comment (flying-sheep) [inline docs/conf.py]: should be `scanpy.pl.DotPlot` and so on, here and everywhere else.

**Review by flying-sheep — CHANGES_REQUESTED** [review]

**Comment by fidelram** [comment]
I give up!

**Comment by flying-sheep** [comment]
You typed `Dotplot`, not `DotPlot` :smile:

File: scanpy/plotting/_dotplot.py [hunk 2]
```diff
@@ -797,39 +811,55 @@ def dotplot(
...
 
     Returns
     -------
-    If `return_fig` is `True`, returns a :class:`DotPlot` object,
+    If `return_fig` is `True`, returns a :class:`~scanpy.pl.DotPlot` object,
     else if `show` is false, return axes dict
 
+    See also
+    --------
+    :class:`~scanpy.pl.Dotplot`: The DotPlot class can be used to to control
```
Comment (flying-sheep) [inline scanpy/plotting/_dotplot.py]:
```suggestion
    :class:`~scanpy.pl.DotPlot`: The DotPlot class can be used to to control
```

File: scanpy/plotting/_stacked_violin.py [hunk 3]
```diff
@@ -544,36 +550,43 @@ def stacked_violin(
...
-    If `return_fig` is `True`, returns a :class:`StackedViolin` object,
+    If `return_fig` is `True`, returns a :class:`~scanpy.pl.StackedViolin` object,
     else if `show` is false, return axes dict
 
+    See also
+    --------
+    :class:`~scanpy.pl.StackedViolin`
+
+    :func:`~scanpy.pl.rank_genes_groups_stacked_violin` to plot marker genes identified
+    using the :func:`~scanpy.tl.rank_genes_groups` function.
```
Comment (flying-sheep) [inline scanpy/plotting/_stacked_violin.py]:
```suggestion
        using the :func:`~scanpy.tl.rank_genes_groups` function.
```

**Review by flying-sheep — CHANGES_REQUESTED** [review]
That should do it

**Comment by fidelram** [comment]
@flying-sheep Thanks! I will take care of the other errors.

**Review by flying-sheep — APPROVED** [review]

File: scanpy/plotting/_stacked_violin.py [hunk 4]
```diff
@@ -544,36 +551,43 @@ def stacked_violin(
...
 
-    The axes used can be obtained using the get_axes() method
+    The axes used can be obtained using the get_axes() method:
+
     >>> axes_dict = vp.get_axes()
 
-    See also
-    --------
-    rank_genes_groups_stacked_violin: to plot marker genes identified using
-    the :func:`~scanpy.tl.rank_genes_groups` function.
```
Comment (flying-sheep) [inline scanpy/plotting/_stacked_violin.py:576]: Now that we know that “see also” sections are definition lists: should this still be deleted?

Comment (fidelram) [reply scanpy/plotting/_stacked_violin.py:576]: It was moved before the examples. But it seems it have an unneeded space.

**Comment by fidelram** [comment]
Now that tests are passing I will replace the outdated doc images.

**Comment by flying-sheep** [comment]
Great, but you can do that in another PR right? this fixes the docs, should go to master ASAP

**Pull request merged**